test/views/partials/messageContentSpec.scala

```diff
@@ -16,46 +16,77 @@
 
   "messsageContent template" must {
+
     "have all information including first read" in new TestClass {
-      val messageContent = new messageContent(layout)(MessageView(
-        "Mike",
-        "sent text",
-        Some("sample text for first read"),
-        "read text",
-        "message body",
-        false)).toString
-
-      messageContent must include("Mike")
-      messageContent must include("sent text")
-      messageContent must include("First read")
+      val messageContent = new messageContent(layout)(
+        MessageView(
+          Some("Mike"),
+          DateTime.parse("2021-02-19T10:29:47.275Z"),
+          Some(DateTime.parse("2021-02-19T10:29:47.275Z")),
+          "message body"))(messagesEn, requestHeader).toString
+
+      messageContent must include("Mike sent this on 19 February 2021 at")
+      messageContent must include("First read on</span> 19 February 2021 at")
+      messageContent must include("First viewed on 19 February 2021 at")
       messageContent must include("message body")
     }
+
     "be handled without first read information" in new TestClass {
       val messageContent =
-        new messageContent(layout)(MessageView("Mike", "sent text", None, "read text", "message body", false)).toString
+        new messageContent(layout)(
+          MessageView(Some("Mike"), DateTime.parse("2021-02-19T10:29:47.275Z"), None, "message body"))(
+          messagesEn,
+          requestHeader).toString
+
+      messageContent must include("Mike sent this on 19 February 2021 at")
+      messageContent mustNot include("First read on")
+      messageContent mustNot include("First viewed on")
+      messageContent must include("message body")
+    }
+
+    "have all information including first read in Welsh" in new TestClass {
+      val messageContent = new messageContent(layout)(
+        MessageView(
+          Some("Mike"),
+          DateTime.parse("2021-02-19T10:29:47.275Z"),
+          Some(DateTime.parse("2021-02-19T10:29:47.275Z")),
+          "message body"))(messagesCy, requestHeader).toString
+
+      messageContent must include("Mike wnaeth anfon y neges hon ar 19 Chwefror 2021 am")
+      messageContent must include("Darllenwyd am y tro cyntaf ar</span> 19 Chwefror 2021 am")
+      messageContent must include("Gwelwyd am y tro cyntaf ar 19 Chwefror 2021 am")
+      messageContent must include("message body")
+    }
+
+    "be handled without first read information in Welsh" in new TestClass {
+      val messageContent =
+        new messageContent(layout)(
+          MessageView(Some("Mike"), DateTime.parse("2021-02-19T10:29:47.275Z"), None, "message body"))(
+          messagesCy,
+          requestHeader).toString
 
-      messageContent must include("Mike")
-      messageContent must include("sent text")
-      messageContent must not include ("First read")
+      messageContent must include("Mike wnaeth anfon y neges hon ar 19 Chwefror 2021 am")
```

Comment:
Do we have space before am for welsh ?

Reply:
This "am" as in "at" but in Welsh, not "am" as "AM/PM"